### ISSUE TITLE
Can't call `setup` for wrapped devices

### ISSUE THREAD

**Issue by blm768** [gleich/is31fl3731#1]
For RGB/matrix devices that are wrapped in a helper struct (such as `Keybow2040`), the fact that the device field isn't `pub` and there's no pass-through method for `setup` means that it's not possible to set the device up. In the short term, just making those fields `pub` is probably the fastest way to fix the issue. In the long run, I've got some ideas for a possible API design that would eliminate the need for a wrapper struct, but that'd be a breaking change.

**Comment by gleich** [comment]
Hey @blm768! Thanks for the bug report. On it now

**Comment by gleich** [comment]
So when you call `configure()` on the struct it will return a IS31FL3731 struct that is configured for the device. You can then call `setup()` on that returned data.

**Comment by blm768** [comment]
That works for most devices, but `Keybow2048::configure` returns a `Keybow2048` `struct` rather than an `IS31FL3731`.

**Comment by gleich** [comment]
yes but you can then just call it on `Keybow2040.device`, right?

**Comment by gleich** [comment]
ah, ok I see. Working on fix now

**Comment by gleich** [comment]
v1.0.1 should be out with the fix. Lemme know if you have any questions/issues. I'd be interested in hearing about your proposed API design, please make a new issue to outline it. Thanks again!

**Comment by blm768** [comment]
That helps, but returning the `IS31FL3731` directly from `configure` makes it a little weird to actually call the `pixel_rgb` method. You basically have to do this:

```rust
let mut device = Keybow2040::configure(i2c);
let mut delay = cortex_m::delay::Delay::new(core.SYST, clocks.system_clock.freq().integer());
device.setup(&mut delay).unwrap();
let mut display = Keybow2040 { device };
display.pixel_rgb(0, 0, 255, 255, 255).unwrap();
```

That does the job for now. Once I get some time, I'll outline some ideas for how to remove the need for a wrapper entirely. Depending on whether you're willing to require const generics, the `width` and `height` fields on `IS31FL3731` could probably go, too (or, rather, become generic parameters rather than in-memory values).

**Comment by gleich** [comment]
ok, I'll take a look at this tomorrow morning

**Comment by gleich** [comment]
Hey @blm768! Sorry for the wait on the follow up. It now just returns `Keybow2040` instead of `IS31FL3731`.  Closing this to continue #2.
